src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/DueDiligenceDataBlockValidator.java

```diff
@@ -18,14 +20,36 @@ public class DueDiligenceDataBlockValidator {
 
      public static final String QUALIFIED_FIELD_NAMES = OverseasEntitySubmissionDto.DUE_DILIGENCE_FIELD + " and " + OverseasEntitySubmissionDto.OVERSEAS_ENTITY_DUE_DILIGENCE;
 
-     public boolean onlyOneBlockPresent(DueDiligenceDto dueDiligenceDto,
-                                        OverseasEntityDueDiligenceDto overseasEntityDueDiligenceDto,
-                                        Errors errors,
-                                        String loggingContext) {
+     public boolean onlyOneCorrectBlockPresent(String whoIsRegistering,
+                                               DueDiligenceDto dueDiligenceDto,
+                                               OverseasEntityDueDiligenceDto overseasEntityDueDiligenceDto,
+                                               Errors errors,
+                                               String loggingContext) {
 
-         return
-         !(bothDueDiligencesPresent(dueDiligenceDto, overseasEntityDueDiligenceDto, errors, loggingContext) ||
-         bothDueDiligencesAbsent(dueDiligenceDto, overseasEntityDueDiligenceDto, errors, loggingContext));
+
+         if (!(bothDueDiligencesPresent(dueDiligenceDto, overseasEntityDueDiligenceDto, errors, loggingContext) ||
+                 bothDueDiligencesAbsent(dueDiligenceDto, overseasEntityDueDiligenceDto, errors, loggingContext))) {
+             return !incorrectDueDiligenceBlockIsPresent(whoIsRegistering, dueDiligenceDto, overseasEntityDueDiligenceDto, errors, loggingContext);
+         } else {
+             return false;
+         }
+    }
+
+    private boolean incorrectDueDiligenceBlockIsPresent(String whoIsRegistering,
+                                                        DueDiligenceDto dueDiligenceDto,
+                                                        OverseasEntityDueDiligenceDto overseasEntityDueDiligenceDto,
+                                                        Errors errors,
+                                                        String loggingContext) {
+
+        boolean isAgentRegistering = whoIsRegistering.equals(AGENT_REGISTERING);
```

Comment:
We need to check if is `agent`, `someone_else` or `null`. This way you can send anything different then agent and it will pass.

Reply:
Have replicated what we do in the web but a null check needs to be added and captured.

Reply:
Yep, agreed, it is just the check on line 48, we need to confirm that if is not the `agent` has to be `someone_else`.

Reply:
Same here, I think we do not need it.